Cognite.Simulator.Utils/RoutineLibraryBase.cs

```diff
@@ -191,21 +200,38 @@ protected virtual V LocalConfigurationFromRoutine(SimulatorRoutineRevision routi
 
         private V ReadAndSaveRoutineRevision(SimulatorRoutineRevision routineRev) {
             
-            V localConfiguration = LocalConfigurationFromRoutine(routineRev);
-            RoutineRevisions.Add(routineRev.Id.ToString(), localConfiguration);
+            V newRevision = LocalConfigurationFromRoutine(routineRev);
+            
+            var result = RoutineRevisions.AddOrUpdate(routineRev.Id.ToString(), newRevision, (key, oldValue) => {
+                if (newRevision.CreatedTime < oldValue.CreatedTime) {
+                    return oldValue;
+                }
+                return newRevision;
+            });
 
-            return localConfiguration;
+            return result;
         }
 
-        private async Task ReadRoutineRevisions(CancellationToken token)
+        private async Task ReadRoutineRevisions(bool init, CancellationToken token)
         {
+            string timeRange = LibState.DestinationExtractedRange.IsEmpty ? "Empty" : LibState.DestinationExtractedRange.ToString();
+            _logger.LogDebug("Updating routine library. There are currently {Num} routine revisions. Extracted range: {TimeRange}",
```

Comment:
Should we also add the `Init` flag to the debug message?

Reply:
happens only once. will clutter the logs

Reply:
sure